webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/Breadcrumb.js

```diff
@@ -33,18 +46,25 @@ const Breadcrumb = ({
 Breadcrumb.propTypes = {
   children: PropTypes.node,
   title: PropTypes.bool,
+  titleReplacement: PropTypes.string,
   isTitle: PropTypes.bool,
   items: PropTypes.arrayOf(PropTypes.shape({
-    caption: PropTypes.string.isRequired,
+    caption: PropTypes.oneOfType([
+      PropTypes.string.isRequired,
+      PropTypes.shape({ icon: PropTypes.string, text: PropTypes.string }),
+    ]),
     url: PropTypes.string,
   })),
+  readOnly: PropTypes.bool,
```

Comment:
[updateFQDN()](https://github.com/theforeman/foreman/blob/a87cf4f78434b7298a60269e2624ac8d1942f0c9/app/assets/javascripts/host_edit_interfaces.js#L317) also alters the title in edit host page, therefore I use a read only prop, so the title in edit host page won't be changed.
an alternative solution for this is the change the logic in legacy javascript

Reply:
why not only trigger the update function on the create page then instead of adding a whole bunch of logic in react as to which titles can or cannot change?